File: 459. Repeated Substring Pattern/459.go
package leetcode

import(
	"strings"
)

func repeatedSubstringPattern(s string) bool {
    for i:=len(s)/2; i>0; i--{
		if len(s)%i == 0{
			start := s[:i]
			tmp := ""
			for j:=0; j<len(s)/i; j++{
				tmp += start
			
				if tmp == s{
					return true
				}
			}
		}
	}
	return false
}

func repeatedSubstringPattern1(s string) bool {
	if len(s) == 0 {
		return false
	}
	ss := (s + s)[1 : len(s)*2-1]
	return strings.Contains(ss, s)
}

File: 459. Repeated Substring Pattern/_459.py
class Solution:
    def repeatedSubstringPattern(self, s: str) -> bool:
        for i in range(len(s)//2,0,-1):
            if len(s)%i == 0 and s[:i]*(len(s)//i) == s:
                return True
        
        return False
    
    def repeatedSubstringPattern1(self, str):
        if not str:
            return False
            
        ss = (str + str)[1:-1]
        return ss.find(str) != -1

        # 和上面一样，但这个更python
        # ss = (str*2)[1:-1]
        # return str in ss

a = Solution()
print(a.repeatedSubstringPattern("aa"))